mm2src/adex_cli/src/adex_proc/adex_proc.rs

```diff
@@ -0,0 +1,196 @@
+            .build();
+
+        match self.transport.send::<_, CoinInitResponse, Json>(command).await {
+            Ok(Ok(ref ok)) => self.response_handler.on_enable_response(ok),
+            Ok(Err(err)) => self.response_handler.print_response(err),
+            Err(err) => {
+                error!("Failed to enable asset: {asset}, error: {err:?}");
+                Err(())
+            },
+        }
+    }
+
+    pub async fn get_balance(&self, asset: &str) -> Result<(), ()> {
+        info!("Getting balance, coin: {asset} ...");
+        let command = Command::builder()
+            .method(Method::GetBalance)
+            .flatten_data(json!({ "coin": asset }))
+            .userpass(self.config.rpc_password())
+            .build();
+
+        match self.transport.send::<_, BalanceResponse, Json>(command).await {
+            Ok(Ok(balance_response)) => self.response_handler.on_balance_response(&balance_response),
+            Ok(Err(err)) => self.response_handler.print_response(err),
+            Err(err) => {
+                error!("Failed to get balance: {err:?}");
+                Err(())
+            },
+        }
+    }
+
+    pub async fn get_enabled(&self) -> Result<(), ()> {
+        info!("Getting list of enabled coins ...");
+
+        let command = Command::<i32>::builder()
+            .method(Method::GetEnabledCoins)
+            .userpass(self.config.rpc_password())
+            .build();
+
+        match self
+            .transport
+            .send::<_, KmdWalletRpcResult<GetEnabledResponse>, Json>(command)
+            .await
+        {
+            Ok(Ok(ok)) => self.response_handler.on_get_enabled_response(&ok),
+            Ok(Err(err)) => self.response_handler.print_response(err),
+            Err(err) => {
+                error!("Failed to get enabled coins: {:?}", err);
+                Err(())
+            },
+        }
+    }
+
+    pub async fn get_orderbook(&self, base: &str, rel: &str, orderbook_config: OrderbookConfig) -> Result<(), ()> {
+        info!("Getting orderbook, base: {base}, rel: {rel} ...");
+
+        let command = Command::builder()
+            .userpass(self.config.rpc_password())
+            .method(Method::GetOrderbook)
+            .flatten_data(OrderbookRequest {
+                base: base.into(),
+                rel: rel.into(),
+            })
+            .build();
+
+        match self.transport.send::<_, OrderbookResponse, Json>(command).await {
+            Ok(Ok(ok)) => self
+                .response_handler
+                .on_orderbook_response(ok, self.config, orderbook_config),
+            Ok(Err(err)) => self.response_handler.print_response(err),
+            Err(err) => {
+                error!("Failed to get orderbook: {err:?}");
+                Err(())
+            },
+        }
+    }
+
+    pub async fn sell(&self, order: SellBuyRequest) -> Result<(), ()> {
+        info!(
+            "Selling: {} {} at price: {} {} for 1 {}",
```

Comment:
pairs can be better written as {}/{}. Just a suggestion, this message can be more informative if it's like `"Selling 20 KMD for 2 BTC at the price of 0.1 BTC per KMD"`

Reply:
done